### ISSUE TITLE
missing document on IMF store and figures

### ISSUE THREAD

**Issue by myrmoteras** [gsautter/goldengate-imagine#117]
this is a doc we need for the demo on Friday in Paris.

there is no way we can find this document on the SRS and download, nor are the images on BLR

`<document ID-DOI="10.5852/ejt.2015.132" ID-ZooBank="EEA49B2B-FD41-4DB0-A34E-573D96EE6E25" _generate="retained" checkinTime="1486388151184" checkinUser="jeremy" docAuthor="Torbjörn Kronestedt" docDate="2015" docId="9E5287A7A20BFFC9DF5D0F05FEB1FC7A" docLanguage="en" docName="Kronestedt, T. (2015). Species of Wadicosa (Araneae, Lycosidae) transfer of two species from Pardosa and description of three new species from Africa.(current) .pdf.imf" docOrigin="European Journal of Taxonomy 132" docSource="http://www.europeanjournaloftaxonomy.eu/index.php/ejt/article/view/254/0" docTitle="Wadicosa cognata Kronestedt, 2015, sp. nov." docUuid="18D4AAA2-7861-4AE8-A820-482BB97CB3B1" docUuidSource="ZooBank" lastPageId="7" lastPageNumber="8" masterDocId="626BFFDFA20EFFCEDD210E0EFFB9FFB1" masterDocTitle="Species of Wadicosa (Araneae, Lycosidae): transfer of two species from Pardosa and description of three new species from Africa" masterLastPageNumber="19" masterPageNumber="1" pageId="5" pageNumber="6" typeStatus="other material" updateTime="1486388151184" updateUser="jeremy">`

http://tb.plazi.org/GgServer/summary/626BFFDFA20EFFCEDD210E0EFFB9FFB1

**Comment by myrmoteras** [comment]
There are also no images linked, that is the doc is not in BLR.

**Comment by myrmoteras** [comment]
it seems that these are also missing in the download, that is I can't get it
--The Thomisidae and Philodromidae (Arachnida: Araneae) of the
Galápagos Islands (Ecuador) masterDocId="FE6CFF98F172FFF16F7BFFB5FFD3FFA2"
--Three new generic records and descriptions of four new species
of jumping spiders (Araneae, Salticidae) from Sri Lanka 
masterDocId="3338FFBDA367FFBA6917D778FFAFFFA2"

**Comment by gsautter** [comment]
I just tried to manually trigger a Zenodo sync of document 626BFFDFA20EFFCEDD210E0EFFB9FFB1 via the server console ... the response is "Document does not exist" ... as it does exist in SRS (see http://tb.plazi.org/GgServer/summary/626BFFDFA20EFFCEDD210E0EFFB9FFB1 ), I conjecture that it was uploaded from GGI directly to our XML Article Repo via "Export -> Export Document to GoldenGATE Server".

In that case, there is no IMF that any synchronizer could put any figures upon Zenodo from ... and of course none you could possibly find and open in GGI ...

To be able to take any further action on this one, Jeremy has to upload the original IMF to the server, i.e., USE "File -> Save Document to GoldenGATE Server", NOT "Export -> Export Document to GoldenGATE Server".

Of course, I could re-convert the PDF to get the figures, but that would lack all the markup Jeremy did. Plus, the server would refuse the write-through from the IMF Repo to the XML Article Repo on the grounds that the document already exists in the latter. That only works if Jeremy does the upload with his login, and even then I might have to assist a little via the console.

Best,
Guido

**Comment by gsautter** [comment]
Could well be the case that the the same applies to the other two documents.

**Comment by myrmoteras** [comment]
But why then do the treatments o those articles show up in TB? If you look at their XML, they even have box elements meaning they are IMF files!?

**Comment by gsautter** [comment]
The XML exports from an IMF just the very same from GGI and from the IMF Store ... and both are just the same pushed to the XML Article Repo, just from different origins. The IMF -> XML conversion even uses pretty much the same code, it just happens in different places.

The XMLs are on TB, just the IMFs are not. That doesn't make any kind of difference to which XML _treatments_ are available.
It does make a whole lot of a difference to the Zenodo Uploader, though, as the latter works on the IMF (little chance of extracting a PNG image from an XML ...), and the resulting HTTP URIs are written through to the XMLs from there.

Bottom line: as long as we have the XMLs, the treatments roper show in TB jus fine (which used to be the entirety of our server before there was GGI), but it takes the IMFs to be on the server as well for any PDFs or images to show up on Zenodo.

**Comment by myrmoteras** [comment]
ok - Jeremy uploaded the file, 626BFFDFA20EFFCEDD210E0EFFB9FFB1 and it is fine, with images on BLR. However, we still wait to see that the figure citations are highlighted and show up in TB. e.g. http://tb.plazi.org/GgServer/html/9E5287A7A20BFFC9DF5D0F05FEB1FC7A 
The figureCitations in the XML include the httpUri

**Comment by myrmoteras** [comment]
here is another EJT I just processed to make sure the process is ok. 43 pages, 32 minutes processing

http://treatment.plazi.org/GgServer/summary/FF95D776EE64FFA41D0B7A02FFDDFFD8

Unlike the doc above, this provides immediate access to the images
the male symbol could be fixed, after it first has misidentified before

**Comment by gsautter** [comment]
As stated above, writing document 626BFFDFA20EFFCEDD210E0EFFB9FFB1 through from IMF Store to the XML Article Repo still requires some coercion from my end, which is why the figure citations have HTTP URIs in the IMF, and the figures are on Zenodo, but those HTTP URIs are not written through to the XML just yet.

This is manly a safety feature, so a newly uploaded IMF doesn't end up overwriting an XML with detailed markup with an almost naked version. Thus the IMF Store requires admin action to overwrite an existing XML that it did not create itself.

Done now, should show soon - if not, try `?cacheControl=force` before opening another ticket, please.

**Comment by gsautter** [comment]
By the way, 32 minutes for 43 pages are quite impressive, especially without a style template. Breaking the minute per page even in generic mode now ... love it :+1:
